Source/objects.cpp

```diff
@@ -5220,26 +5206,29 @@ void ObjChangeMapResync(int x1, int y1, int x2, int y2)
 	}
 }
 
-void TryDisarm(int pnum, int i)
+void TryDisarm(Player &player, int i)
 {
-	if (pnum == MyPlayerId)
+	if (&player == MyPlayer)
 		NewCursor(CURSOR_HAND);
-	if (!Objects[i]._oTrapFlag) {
+
+	Object &target = Objects[i];
+
+	if (!target._oTrapFlag) {
 		return;
 	}
-	int trapdisper = 2 * Players[pnum]._pDexterity - 5 * currlevel;
+	int trapdisper = 2 * player._pDexterity - 5 * currlevel;
 	if (GenerateRnd(100) > trapdisper) {
 		return;
 	}
 	for (int j = 0; j < ActiveObjectCount; j++) {
 		Object &trap = Objects[ActiveObjects[j]];
 		if (trap.IsTrap() && dObject[trap._oVar1][trap._oVar2] - 1 == i) {

```

Comment:
```suggestion
		if (trap.IsTrap() && ObjectAtPosition({trap._oVar1, trap._oVar2}) == &target) {
```

Would let you take an object reference in this function.